appservice/consumers/roomserver.go

```diff
@@ -151,15 +152,25 @@ func (s *OutputRoomEventConsumer) onMessage(
 		return true
 	}
 
+	txnID := ""
```

Comment:
`msgs` can't be 0 at this point, we already iterated over them and did a `len(events) == 0` check before. So this can be removed.
I'd prefer having
```go
txnID := fmt.Sprintf("%d_%d", events[0].Event.OriginServerTS(), len(transaction))
```
from `sendEvents` here (instead of `txnID := ""`), so it's easier to read.

Reply:
Fair enough.